src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/ReadyToRunSingleAssemblyCompilationModuleGroup.cs

```diff
@@ -210,5 +211,30 @@ public override bool GeneratesPInvoke(MethodDesc method)
 
             return !Marshaller.IsMarshallingRequired(method);
         }
+
+        public override bool TryGetModuleTokenForExternalType(TypeDesc type, out ModuleToken token)
+        {
+            Debug.Assert(!VersionsWithType(type));
+
+            if (_typeRefsInCompilationModuleSet == null)
+            {
+                _typeRefsInCompilationModuleSet = new Dictionary<TypeDesc, ModuleToken>();
+
+                foreach (var module in _compilationModuleSet)
+                {
+                    EcmaModule ecmaModule = (EcmaModule)module;
+                    foreach (var typeRefHandle in ecmaModule.MetadataReader.TypeReferences)
+                    {
+                        TypeDesc typeFromTypeRef = ecmaModule.GetType(typeRefHandle);
```

Comment:
It feels like you need a try/catch here in case the typeref isn't resolvable.

Reply:
Fixed